rfcs/0003-rails-assets.md

```diff
@@ -0,0 +1,258 @@
+# Rails Asset Pipeline Support
+
+## Proposal
+
+Include a Rails assets buildpack in the Ruby buildpack family. The buildpack
+will be charged with precompiling assets so that they can be accessed in a
+production deployment. The buildpack will be included in each of the
+"webserver" buildpack groups, but not the Rake group. To support the buildpack
+build process, the `node-engine` buildpack will also be included in any group
+that includes the new buildpack.
+
+## Motivation
+
+[Rails](https://rubyonrails.org/) is a popular web framework used in the Ruby
+community. In recent versions of Rails, Javascript, CSS, and other front-end
+assets are managed in a build system called the [Asset
+Pipeline](https://guides.rubyonrails.org/asset_pipeline.html).
+
+Rails developers that use the Asset Pipeline will expect that their assets are
+precompiled and made available in the built image. This means that the Ruby
+buildpack groups should include a buildpack that executes this process to
+precompile assets when it is needed.
+
+## Implementation
+
+A new `rails-assets` buildpack will be developed to detect that the Rails Asset
+Pipeline is present, and then precompile those assets so that they can be
+included in the built image.
+
+### Detection Criteria
+
+The buildpack will detect if the `Gemfile` contains the `rails` gem and the
+`app/assets` directory is present. According to the
+[documentation](https://guides.rubyonrails.org/asset_pipeline.html#how-to-use-the-asset-pipeline),
+the `app/assets` directory is the location that a Rails application will expect
+to find assets that need to be compiled before a production deployment.
+
+If the buildpack detects, then it will need to require `node` since a Node.js
+runtime is required to execute the asset precompilation process.
+
+### Build Process
+
+The build process of the buildpack will execute a precompilation process that
+generates a set of assets to be served in a production deployment.  The
+[documentation](https://guides.rubyonrails.org/asset_pipeline.html#precompiling-assets)
+outlines that assets can be precompiled by executing `bundle exec rails
+assets:precompile`.
```

Comment:
Just curious, would this give us some way to cache the static asset in subsequent builds?

Reply:
Yes. I have not outlined caching of assets to that level of detail, but presumably providing a cache of the built artifacts or intermediate products would be possible to implement.